src/core/memory.d

```diff
@@ -130,6 +130,8 @@ private
     extern (C) void*   gc_addrOf( void* p ) pure nothrow;
     extern (C) size_t  gc_sizeOf( void* p ) pure nothrow;
 
+    extern (C) void gc_usage( out size_t used, out size_t free) nothrow;
```

Comment:
> How about introducing new core.gcapi module that can be imported by both GC implementations and core.memory? 

That's more or less what gc.gcinterface does in #1581.

> This whole business of manually maintaining extern(C) declarations in sync feels like legacy.

I agree. If the extern(C) interface is to stay (it is barely useful to begin with), it should be opt-in provided by a GC proxy. Having the calls go through an extern(C) interface also inhibits some inlining for the common case.